internal/pkg/cli/svc_init.go

```diff
@@ -805,10 +807,24 @@ func validateWorkspaceApp(wsApp, inputApp string, store store) error {
 
 func (o initSvcOpts) convertStringsToAssets(sources []string) ([]manifest.FileUpload, error) {
 	assets := make([]manifest.FileUpload, len(sources))
+	if o.wsPendingCreation {
```

Comment:
since we don't _know_ that our current location is the eventual project root, should we just skip validation in this case? since it'll get validated on `svc deploy` anyways

Reply:
Good question. I re-thought the process, and realized a few things:
1. When there is no workspace, like you said, we should leave the validation to `svc deploy`. Realistically, `Validate` and `Ask` without a workspace is a scenario that's only happens during `copilot init`. 
2. When there is a workspace, we should validate the path _during_ the prompts, not after, when we convert the path to struct. If we validate during the prompts, folks can fix their path immediately without having to restart the whole command again.